### ISSUE TITLE
[PATCH] requestReceived drops arguments after two or more '?' in the uri

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11085]
|<img alt="zectbumo's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| zectbumo reported|
|-|-|
|Trac ID|trac#1550|
|Type|defect|
|Created|2006-03-06 19:49:10Z|

An uri with two or more '?' will get dropped and a log.msg("May ignore parts of this invalid URI: %s" % repr(self.uri)) would be logged.
The url parser should be friendly and accept multiple '?' even though the uri is invalid.  This is done by uri.split('?', 1) instead of uri.split('?').

Attachments:

* [twisted.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/3c8/3c8101acb51e8f51363933e63bfb9106ec64d6e4/d28a367b58eba790fb131d7a60d9534697fcdd09.patch) (705 bytes) - added by zectbumo on 2006-03-10 19:47:03Z - 
* [twistedtest.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/3c8/3c8101acb51e8f51363933e63bfb9106ec64d6e4/8d7282df9f87324bcdf3f34d7d109813e94f766f.patch) (813 bytes) - added by zectbumo on 2006-03-10 19:47:20Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__1550 1550
type__defect defect
reporter__zectbumo zectbumo
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1141674550000000 1141674550000000
changetime__1159029252000000 1159029252000000
version__2_0 2.0
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)|@jyknight commented|
|-|-|

Do you have a reason to believe this is a good interpretation? If we're going to have a behavior beyond what the RFC says, it should at least be based on what other servers do.

**Comment by twisted-trac** [comment]
|<img alt="zectbumo's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|zectbumo commented|
|-|-|

I have tried apache/mod_python:Publisher and tomcat and they both pass the ? in as the argument.

**Comment by twisted-trac** [comment]
|<img alt="zectbumo's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|zectbumo commented|
|-|-|

ready for review

**Comment by twisted-trac** [comment]
|<img alt="zectbumo's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|zectbumo commented|
|-|-|

updated title [PATCH]

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

It'd be nice to leave the test for a _valid_ URI alone and add a new test for this lenient handling of incorrect URIs.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun removed owner|
|-|-|

Ready for review in invalid-qmark-1550

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_7"></a>|@jml set owner to @exarkun|
|-|-|

Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [18268]) Merge invalid-qmark-1550

Author: exarkun, zectbumo
Reviewer: jml
Fixes [#11085](https://github.com/twisted/twisted/issues/11085)

Add support for invalid URLs which contain more than one '?'.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
